misk/src/test/kotlin/misk/digester/SlidingWindowDigestTest.kt

```diff
@@ -0,0 +1,232 @@
+        newWindowDigest(windowsT0_2[2], listOf(10.0))
+    ))
+  }
+
+  @Test
+  fun testSlidingWindowDigestMergeInValuesToValues() {
+    val src = newSlidingWindowDigestTest()
+    val dest = newSlidingWindowDigestTest()
+    val windowsT0_2 = windows(src)
+    src.observe(100.0) // in t0-t2 buckets
+    val windowsT3_5 = advanceWindows(3, src)
+    src.observe(200.0) // in t3-t5 buckets
+    advanceWindows(3, src)
+    advanceWindows(1, dest)
+    dest.observe(10.0) // in t1-t3 buckets
+    Assertions.assertThat(dest.openDigests(false).count()).isEqualTo(3)
+    dest.utcNowClock = src.utcNowClock
+    dest.mergeIn(src.closedDigests(windowsT0_2[0].end))
+    expectWindowDigests(dest.windows, listOf(
+        newWindowDigest(windowsT0_2[0], listOf(100.0)),
+        newWindowDigest(windowsT0_2[1], listOf(10.0, 100.0)),
+        newWindowDigest(windowsT0_2[2], listOf(10.0, 100.0)),
+        newWindowDigest(windowsT3_5[0], listOf(10.0, 200.0)),
+        newWindowDigest(windowsT3_5[1], listOf(200.0)),
+        newWindowDigest(windowsT3_5[2], listOf(200.0)
+        )))
+  }
+
+  @Test
+  fun testSlidingWindowDigestGC() {
+    val digest = newSlidingWindowDigestTest()
+    digest.observe(10.0)
+    // Move just past the threshold for collecting the last window
+    digest.utcNowClock = Clock.fixed(windows(digest)[2].end.plusMinutes(1).plusNanos(1).toInstant(),
+        ZoneId.of("UTC"))
+    digest.observe(20.0)
+    val windows = windows(digest)
+    expectWindowDigests(digest.windows, listOf(
+        newWindowDigest(windows[0], listOf(20.0)),
+        newWindowDigest(windows[1], listOf(20.0)),
+        newWindowDigest(windows[2], listOf(20.0))
+    ))
+  }
+
+  fun setClock(t: ZonedDateTime, digest: SlidingWindowDigest) {
+    require(!t.toInstant().isBefore(digest.utcNowClock.instant())) {
+      "Cannot go back in time"
+    }
+
+    digest.utcNowClock = Clock.fixed(t.toInstant(), ZoneId.of("UTC"))
+  }
+
+  fun windows(slidingWindow: SlidingWindowDigest): List<Window> {
+    return slidingWindow.windower.windowsContaining(
+        ZonedDateTime.ofInstant(slidingWindow.utcNowClock.instant(), ZoneId.of("UTC")))
+  }
+
+  fun advanceWindows(n: Int, digest: SlidingWindowDigest): List<Window> {
+    repeat(n) {
+      setClock(windows(digest)[0].end, digest)
+    }
+    return windows(digest)
+  }
+
+  fun newWindowDigest(window: Window, values: List<Double>): WindowDigest {
+    return WindowDigest(
+        window,
+        FakeDigest(values)
+    )
+  }
+
+  fun newSlidingWindowDigestTest(): SlidingWindowDigest {
+    return SlidingWindowDigest(
+        baseClock,
+        Windower(10, 3),
+        fun() = FakeDigest()
+    )
+  }
+
+  fun expectWindowDigests(actual: List<WindowDigest>, expected: List<WindowDigest>) {
+    Assertions.assertThat(expected.count()).isEqualTo(actual.count())
+    for (i in 0 until actual.count()) {
+      if (i >= expected.count()) {
+        break
+      }
+      //Compare window of each WindowDigest
```

Comment:
Nit: Comments should have spaces: `// Compare` instead of `//Compare` (or `* xyz` instead of `*xyz`)